src/juju/watchers.ts

```diff
@@ -39,19 +45,79 @@ export function processDeltas(
     return;
   }
   deltas.forEach((delta) => {
-    // The delta messages are in the format [entity, action, data].
+    // Delta is in the format [entityType, actionType, data].
     const modelUUID = delta[2]["model-uuid"];
+    if (!modelWatcherData[modelUUID]) {
+      modelWatcherData[modelUUID] = generateModelWatcherBase();
+    }
     switch (delta[0]) {
+      case "action":
+        switch (delta[1]) {
```

Comment:
Nested switch feel wrong and will grow messy. Why not check and switch the `entityType` function handlers with switch/case?

Reply:
I actually thought the exact same thing :) As there are a few other delta types to add I was going to wait until those ones are completed and then look for a way to clean this up. I expect that I'll have a couple utility functions that are called instead of this repetition.